### PR TITLE
Adding docstrings

### PR DESCRIPTION
**Pull request by brookwander**

### CODE REVIEW

**Review by hmera5 — COMMENTED** [review]
Needs review/additional information: 
#############################
**adslab_predictions**

8. (to) –

gpu_mem_per_sample (float, optional) – The approximate memory used by a single

relaxations (adslab during) –

names. –

batch_size (int, optional) – The number of adslabs loaded onto a gpu at a single

##############################
**cache_utils**
memmaps (and numpy) –

shard_digits (int, optional) – Generate 16^(shard digits) different folders to

2. (store functions in. Defaults to) –

###############################
**dask_kube_utils**
cluster (dask.distributed.client.Client) – a dask cluster to run code on.

config_path (str) – a file path containing a config yml file defining the

worker (specifications of the new) –

#################################
**enumerate_slabs_adslabs**
class catlas.enumerate_slabs_adslabs.CustomAdsorbate(ads_atoms, bond_indices, smiles)
A custom adsorbate object for ocdata **compatability.**

class catlas.enumerate_slabs_adslabs.CustomBulk(bulk_atoms)
A custom bulk object for **ocdata compatability.**

**Comment by brookwander** [comment]
@hmera5 can you be more specific? Whats unclear? ty

**Comment by hmera5** [comment]
> @hmera5 can you be more specific? Whats unclear? ty

I listed the items that define the parameters but some of the definitions are not finished, and some may have missing descriptions, and some have minor typos.

**Comment by hmera5** [comment]
> @hmera5 can you be more specific? Whats unclear? ty

I provided some examples for docstring or section in the link

**Comment by KirbyBroderick** [comment]
Could you comment directly on the code? (hit the "files changed" tab.) It's hard to follow the conversation in this format

**Comment by hmera5** [comment]
> @hmera5 Could you comment directly on the code? (hit the "files changed" tab.) It's hard to follow the conversation in this format

Ok, I'll try

File: catlas/nuclearity.py [hunk 1]
```diff
@@ -65,12 +65,11 @@ def get_nuclearity_neighbor_counts(surface_atoms_of_element, connectivity_matrix
     """
     Function that counts the like surface neighbors for surface atoms
     Args:
-        surface_atoms_of_element: list of all surface atoms which are of a specific
-        element
-        connectivity_matrix: matrix memorializing which atoms in the slab are connected
+        surface_atoms_of_element (ase.Atoms): list of all surface atoms which are of a specific element
+        connectivity_matrix (numpy.ndarray[int8]): which atoms in the slab are connected
```
Comment (KirbyBroderick) [inline catlas/nuclearity.py:69]: Do you happen to know if the `type` of connectivity/nuclearity objects are correct as listed? I got lost in documentation and made an educated guess.

File: catlas/nuclearity.py [hunk 2]
```diff
@@ -105,12 +102,12 @@ def evaluate_infiniteness(hist, hist_rep):
...
     """
-    Function that gets a connectivity matrix by looking at nearest neighbors.
+    Get connectivity matrix by looking at nearest neighbors.
     Args:
-        slab_atoms: atoms object of the slab
+        slab_atoms (ase.Atoms): a slab object
 
     Returns:
-        overall_connectivity_matrix
+        numpy.ndarray[int8]: an array describing what atoms are connected
```
Comment (KirbyBroderick) [inline catlas/nuclearity.py:110]: Is this correct? Is the array 0 if no connection, 1 if connection?

File: documentation/advanced_use.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# Advanced Use
+
+WIP
+
+Current ideas for what to include:
```
Comment (KirbyBroderick) [inline documentation/advanced_use.md:5]: 4. Making your own cluster
5. Running on NERSC
6. How to parse/understand outputs?

File: documentation/intro.md [hunk 4]
```diff
@@ -2,7 +2,7 @@
...
...oject/Open-Catalyst-Dataset) that was built out for the OC20 dataset[@OC20] to enumerate adsorbate-surface configurations. The workflow starts from relaxed bulk structures. The bulk structures which were included in OC20 are included in the repo in the `catlas/bulk_structures` folder. This set of bulks was originally taken from the Materials Project Database[@MP] and then relaxed with RPBE. Optionally you can add your own bulk file. For more info, see advanced use. Slabs are cut from the bulk structures using pymatgen[@pymatgen]. From there, adsorbates are placed on the slabs using CatKit[@catkit]. Adsorbates from OC20 are included here in the `catlas/adsorbate_structures` folder. Optionally you can add your own adsorbate file. For more info, see advanced use. Finally, ML inference is performed using the [infrastructure](https://github.com/Open-Catalyst-Project/ocp) for OCP models. To expedite this process, catlas parallelizes enumeration and inference over many CPUs and GPUs using dask[@dask]. Pretrained model checkpoint files may be downloaded [here](https://github.com/Open-Catalyst-Project/ocp/blob/main/MODELS.md), but have been ommited from this repo because of their file size.
```
Comment (KirbyBroderick) [inline documentation/intro.md:5]: Do other people have access to our bulk_structures folder and models? Would they have to download and link them manually or will they be downloaded with catlas?

File: documentation/making_a_config.md [hunk 5]
```diff
@@ -51,36 +51,62 @@ input_options:
...
+### 3. Bulk ids to ignore:
+a list of bulk ids to exclude
 ```filter_ignore_bulk_ids: ['mp-126','mp-30', 'mp-81', 'mp-13', 'mp-79']```
-4. Number of elements: a list of the numbers which are acceptable i.e. `[2, 3]` gives you binary and ternary materials
+### 4. Number of elements:
+a list of the numbers which are acceptable i.e. `[2, 3]` gives you binary and ternary materials
 ```filter_by_num_elements: [2, 3]```
-5. Required elements: a list of elements which must be in each material. If you say "Cu", then all copper alloys will be considered.
+### 5. Required elements:
+a list of elements which must be in each material. If you say "Cu", then all copper alloys will be considered.
```
Comment (KirbyBroderick) [inline documentation/making_a_config.md:83]: "all" -> "only"

File: documentation/making_a_config.md [hunk 6]
```diff
@@ -51,36 +51,62 @@ input_options:
...
 ```filter_by_required_elements: ["Cu"]```
-6. Number of atoms: this is useful to avoid very large structures which are costly to compute. It filters out all unit cell structures with more atoms than the number specified.
+### 6. Number of atoms:
+this is useful to avoid very large structures which are costly to compute. It filters out all unit cell structures with more atoms than the number specified.
```
Comment (KirbyBroderick) [inline documentation/making_a_config.md:86]: Might mention that people can use the Sankey diagram to calibrate this value? Or maybe people running this will have some intuition

File: documentation/making_a_config.md [hunk 7]
```diff
@@ -51,36 +51,62 @@ input_options:
...
   active: ["Pd", "Ag", "Cu"]
   host: ["Zn"]
 ```
-8. Filter by element groups: A list of periodic table groups which are of interest. The groups should be specified as a list of any of the following: `["transition metal", "post-transition metal", "metalloid", "rare earth metal", "alkali", "alkaline", "alkali earth", "chalcogen", "halogen"]`
+### 8. Filter by element groups:
+A list of periodic table groups which are of interest. The groups should be specified as a list of any of the following: `["transition metal", "post-transition metal", "metalloid", "rare earth metal", "alkali", "alkaline", "alkali earth", "chalcogen", "halogen"]`
 ```filter_by_element_groups: ["transition metal"]```
-9. Energy above hull: a float of the maximum energy above hull you would like to consider
+### 9. Energy above hull:
+a float of the maximum energy above hull you would like to consider
```
Comment (KirbyBroderick) [inline documentation/making_a_config.md:99]: Maybe include units?

File: documentation/making_a_config.md [hunk 8]
```diff
@@ -51,36 +51,62 @@ input_options:
...
 ```filter_by_bulk_e_above_hull: 0.05```
-10. Band gap: This requires one or both of the minimum and maximum band gap you would like to be considered to be specified. If both are, then materials in the range will be filtered for. If only the maximum is, then anything with a band gap less than the value given will be filtered for. If only the minimum is, then anything with a band gap greater than the value given will be filtered for.
+### 10. Band gap:
```
Comment (KirbyBroderick) [inline documentation/making_a_config.md:101]: Probably include units

File: documentation/making_a_config.md [hunk 9]
```diff
@@ -51,36 +51,62 @@ input_options:
...
 ```
 filter_by_bulk_band_gap:
   min_gap: 0.1
   max_gap:0.3
 ```
-11. Pourbaix stability: This is only supported for Materials Project materials as of now. It selects the Pourbaix stable materials under the conditions you specify. Conditions may be specified as a range or as a list of specific values. You may also specify a maximum decoposition energy.  Step size is the increment that will be used for the interval. If the material is stable at any of the conditions considered, it will not be filtered out. A path to an lmdb file containing Pourbaix info for your materials is required. If the file path doesnt exist, the Pourbaix info will be queried from MP as a part of the run. This does take a bit of time. Be careful of having to many workers querying to avoid being black listed.
+### 11. Pourbaix stability:
```
Comment (KirbyBroderick) [inline documentation/making_a_config.md:108]: Units of maximum decomp energy are eV/(metal atom)

File: documentation/making_a_config.md [hunk 10]
```diff
@@ -51,36 +51,62 @@ input_options:
...
+This is only supported for Materials Project materials as of now. It selects the Pourbaix stable materials under the conditions you specify. Conditions may be specified as a range or as a list of specific values. You may also specify a maximum decoposition energy.  Step size is the increment that will be used for the interval. If the material is stable at any of the conditions considered, it will not be filtered out. A path to an lmdb file containing Pourbaix info for your materials is required. If the file path doesnt exist, the Pourbaix info will be queried from MP as a part of the run. This does take a bit of time. Be careful of having to many workers querying to avoid being black listed.
```
Comment (KirbyBroderick) [inline documentation/making_a_config.md:109]: Somewhat unclear, could rephrase: ` Conditions may be specified as a list of specific values, or a range with an interval.`

**Review by KirbyBroderick — CHANGES_REQUESTED** [review]
LGTM, few questions/suggestions

File: documentation/intro.md (hunk 4 above)
Comment (brookwander) [reply documentation/intro.md:5]: I think this is covered in install, the bulk file is in the repo, the models they are instructed to download and told where

File: documentation/advanced_use.md (hunk 3 above)
Comment (brookwander) [reply documentation/advanced_use.md:5]: I think 4 & 5 is in use. 6 probably doesnt belong in advanced use but could be added somewhere else

File: catlas/nuclearity.py (hunk 2 above)
Comment (brookwander) [reply catlas/nuclearity.py:110]: haha didnt you add that?

Comment (brookwander) [reply catlas/nuclearity.py:110]: I think its always a matrix though. If there are no connections it will be a matrix of 0s

File: catlas/nuclearity.py (hunk 1 above)
Comment (brookwander) [reply catlas/nuclearity.py:69]: definitely the first is wrong. As the description includes it is a list

File: documentation/making_a_config.md (hunk 5 above)
Comment (brookwander) [reply documentation/making_a_config.md:83]: why?

Comment (brookwander) [reply documentation/making_a_config.md:83]: seems the same to me

File: documentation/making_a_config.md (hunk 6 above)
Comment (brookwander) [reply documentation/making_a_config.md:86]: yeah I think since the sankey diagram is constructed at the end of a run its not actually very useful for this purpose. We could provide a suggested value to get to this point though?

File: documentation/making_a_config.md (hunk 7 above)
Comment (brookwander) [reply documentation/making_a_config.md:99]: seems like a good idea

File: documentation/making_a_config.md (hunk 8 above)
Comment (brookwander) [reply documentation/making_a_config.md:101]: yep

File: documentation/making_a_config.md (hunk 5 above)
Comment (KirbyBroderick) [reply documentation/making_a_config.md:83]: I guess, it just seems clearer that you're limiting it to copper alloys- it would also be clearer if the example were multicomponent

**Review by KirbyBroderick — APPROVED** [review]
LGTM
